File: inetpub/wwwroot/Services/Components/Logistik/Change02sUserControls/AddressSearch.vb
Imports System.ComponentModel

Public Class AddressSearch
    Inherits Controls.AddressSearch

    Public Const Abholadresse As String = "Abholadresse"
    Public Const Auslieferung As String = "Auslieferung"
    Public Const Halter As String = "Halter"

    Private transferPage As ITransferPage

    <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden), Browsable(False)>
    Public Property Address As Adresse

    Protected Overrides Sub OnInit(e As System.EventArgs)
        MyBase.OnInit(e)

        Me.transferPage = DirectCast(Me.Page, ITransferPage)
    End Sub

    Protected Overrides Function DoSearch(addressType As String, name As String, postcode As String, town As String) As System.Data.DataTable
        Me.transferPage.Transfer.FillAdressen(Me.transferPage.CSKUser, Me.Page, addressType, name, postcode, town, Me.transferPage.GetKundennummer())
        Select Case addressType
            Case AddressSearch.Abholadresse
                Return Me.transferPage.Transfer.Abholadresse
            Case AddressSearch.Auslieferung
                Return Me.transferPage.Transfer.Zieladresse
            Case AddressSearch.Halter
                Return Me.transferPage.Transfer.Halter
            Case Else
                Return Nothing
        End Select
    End Function

    Protected Overrides Sub ExtractData(addressRow As System.Data.DataRow)
        Me.Address = New Adresse() With
                     {
                        .DebitorNr = addressRow.Field(Of String)("KUNNR"),
                        .Ansprechpartner = addressRow.Field(Of String)("Name2"),
                        .Name = addressRow.Field(Of String)("NAME1"),
                        .Ort = addressRow.Field(Of String)("ORT01"),
                        .Postleitzahl = addressRow.Field(Of String)("PSTLZ"),
                        .Straße = addressRow.Field(Of String)("STRAS"),
                        .Telefon = addressRow.Field(Of String)("TELNR"),
                        .Land = addressRow.Field(Of String)("LAND1"),
                        .Carport = addressRow.Field(Of String)("POS_TEXT"),
                        .EMail = addressRow.Field(Of String)("Email")
                    }
    End Sub

End Class

File: inetpub/wwwroot/Services/Components/Zulassung/UserControls/AddressSearch.cs
using System;
using System.ComponentModel;
using System.Data;

using CKG.Components.Zulassung.DAL;

namespace CKG.Components.Zulassung.UserControls
{
    public class AddressSearch : Controls.AddressSearch
    {
        private IWizardPage wizardPage;

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        [Browsable(false)]
        public AddressData Address { get; set; }
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        [Browsable(false)]
        public string KundenDebitorNummer { get; set; }

        protected override void OnInit(EventArgs e)
        {
            base.OnInit(e);
            this.wizardPage = (IWizardPage)Page;
        }

        protected override DataTable DoSearch(string addressType, string name, string postcode, string town)
        {
            return this.wizardPage.DAL.SearchAddress(addressType, name, postcode, town);
        }

        protected override void ExtractData(DataRow addressRow)
        {
            this.Address = new AddressData()
            {
                Name1 = addressRow.Field<string>("NAME1"),
                Name2 = addressRow.Field<string>("NAME2"),
                Street = addressRow.Field<string>("STRAS"),
                ZipCode = addressRow.Field<string>("PSTLZ"),
                City = addressRow.Field<string>("ORT01"),
                Description = addressRow.Field<string>("POS_Text"),
            };

            this.KundenDebitorNummer = addressRow.Field<string>("SAPNR");
        }
    }
}